src/main/java/org/opensearch/knn/quantization/models/quantizationParams/ScalarQuantizationParams.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationParams;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.knn.quantization.enums.ScalarQuantizationType;
+
+import java.io.IOException;
+import java.util.Locale;
+
+/**
+ * The ScalarQuantizationParams class represents the parameters specific to scalar quantization (SQ).
+ * This class implements the QuantizationParams interface and includes the type of scalar quantization.
+ */
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor // No-argument constructor for deserialization
+@EqualsAndHashCode
+public class ScalarQuantizationParams implements QuantizationParams {
+    private ScalarQuantizationType sqType;
+    private static final long serialVersionUID = 1L; // Version ID for serialization
+
+    /**
+     * Static method to generate type identifier based on ScalarQuantizationType.
+     *
+     * @param sqType the scalar quantization type.
+     * @return A string representing the unique type identifier.
+     */
+    public static String generateTypeIdentifier(ScalarQuantizationType sqType) {
```

Comment:
why this is a public function of `ScalarQuantizationParams`? who will use this outside of QF? is this function better be present in an utility/helper?

Reply:
As Discussed , we are keeping this now , will Revisit this later , if we need to make some more refactor.